pydicom/dataset.py

```diff
@@ -1746,44 +1833,46 @@ def _pretty_str(self, indent=0, top_level_only=False):
 
         # Display file meta, if configured to do so, and have a non-empty one
         if (
-            hasattr(self, "file_meta")
-            and self.file_meta is not None
-            and len(self.file_meta) > 0
+            hasattr(self, "file_meta") and self.file_meta
             and pydicom.config.show_file_meta
         ):
-            strings.append("Dataset.file_meta -------------------------------")
-            for data_element in self.file_meta:
-                with tag_in_exception(data_element.tag):
-                    strings.append(indent_str + repr(data_element))
-            strings.append("-------------------------------------------------")
-
-        for data_element in self:
-            with tag_in_exception(data_element.tag):
-                if data_element.VR == "SQ":  # a sequence
-                    strings.append(indent_str + str(data_element.tag) +
-                                   "  %s   %i item(s) ---- " %
-                                   (data_element.description(),
-                                    len(data_element.value)))
+            strings.append(f"{'Dataset.file_meta ':-<49}")
+            for elem in self.file_meta:
+                with tag_in_exception(elem.tag):
+                    strings.append(indent_str + repr(elem))
+            strings.append(f"{'':-<49}")
```

Comment:
Did I mention that I like f-strings?

Reply:
Me too! They're a little bit slower than using `.format()` though